Userland/Libraries/LibWeb/HTML/EventLoop/EventLoop.cpp

```diff
@@ -341,4 +367,51 @@ void EventLoop::unregister_environment_settings_object(Badge<EnvironmentSettings
     VERIFY(did_remove);
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#same-loop-windows
+NonnullRefPtrVector<Window> EventLoop::same_loop_windows() const
+{
+    NonnullRefPtrVector<Window> windows;
+    for (auto& document : documents_in_this_event_loop())
+        windows.append(document.window());
+    return windows;
+}
+
+// https://w3c.github.io/hr-time/#dfn-unsafe-shared-current-time
+double EventLoop::unsafe_shared_current_time() const
+{
+    timespec now_spec;
+    clock_gettime(CLOCK_MONOTONIC, &now_spec);
+    return now_spec.tv_sec * 1e3 + now_spec.tv_nsec / 1e6;
```

Comment:
Using AK::Time would make this much nicer

Reply:
Ah, didn't know that existed. I only saw clock_gettime in ElapsedTimer and Temporal.

Reply:
That might actually pre-date AK::Time, I'll look into changing that as well :)